src/checked_runner.ml

```diff
@@ -471,6 +475,10 @@ module Make (Backend : Backend_extended.S) = struct
         let handle = {Handle.var; value= None} in
         let g, a = flatten_as_prover next_auxiliary stack (k handle) in
         ( (fun s ->
+            if !(s.as_prover) then
```

Comment:
This should currently be impossible to hit, but it's unsafe to do anything but fail if we ever get here.

I've added an extra check in `run` to make extra sure this can't be hit, updated the error message, and wrote a comment detailing the reason that we have to hard fail.